test/integration/redirect-getserversideprops-rewrites/test/index.test.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import {
+  renderViaHTTP,
+  findPort,
+  launchApp,
+  killApp,
+  waitFor,
+} from 'next-test-utils'
+import webdriver from 'next-webdriver'
+
+// test suites
+
+const context = {}
+jest.setTimeout(1000 * 60 * 5)
+
+describe('getServerSideProps redirects', () => {
+  beforeAll(async () => {
+    context.appPort = await findPort()
+    context.server = await launchApp(join(__dirname, '../'), context.appPort, {
+      env: { __NEXT_TEST_WITH_DEVTOOL: 1 },
+    })
+
+    // pre-build all pages at the start
+    await Promise.all([
+      renderViaHTTP(context.appPort, '/'),
+      renderViaHTTP(context.appPort, '/some-page'),
+    ])
+  })
+  afterAll(() => killApp(context.server))
+
+  it('should use a client-side navigation for a rewritten URL', async () => {
+    const browser = await webdriver(context.appPort, '/')
+
+    await browser.executeScript(function () {
+      // During a browser navigation global variables are reset,
+      // So by chaking that the __SAME_PAGE variable is still defined
+      // then the client-side navigation has happened
+      window.__SAME_PAGE = true
+    })
+
+    await browser.elementByCss('#link-with-rewritten-url').click()
+
+    // Wait for a potential page reload
+    await waitFor(1000)
+
+    // Wait until the new props are rendered
+    await browser.elementByCss('.refreshed')
+
+    const isSamePage = await browser.executeScript(function () {
+      return window.__SAME_PAGE || false
+    })
+
+    expect(isSamePage).toBe(true)
```

Comment:
We can add this test to the existing `test/integration/gssp-redirect` suite

Reply:
Sure! Out of curiosity, what does `gssp` stand for?

Reply:
Ok I think I got it : `GetServerSideProps` 😛

Reply:
I'm working on it and making good progress. However this change isn't ideal since the `serverless mode` tests are writing content in `next.config.js` during the tests (hence overriding my required configuration).

I would like to suggest an other solution : move this test to a new folder `gssp-redirect-with-rewrites` next to other folders (like `gssp-redirect` and `gssp-redirect-base-path` for instance). What do you think ?  It would also avoid adding context-specific configuration to the `next.config.js` file since those modifications are not required for other tests suites.